### ISSUE TITLE
Please remove the `decorator` dependency

### ISSUE THREAD

**Issue by mindreframer** [madeinussr/swap#1]
It seems not to be in use anymore .... 

:) Thanks for this little useful package!

**Comment by implicitly-awesome** [comment]
Hi, @mindreframer, thank you submitting the issue :) will wipe it out

**Comment by implicitly-awesome** [comment]
done

**Comment by mindreframer** [comment]
thanks! I was playing with `swap` more and noticed that it slows down the execution of tests quite significantly and also does not play well with live code reloaders [Cortex](https://github.com/urbint/cortex)... Not sure if that is helpful feedback, yet I think it is going to be tough to provide a fast, correct, nicely behaving module swapper for Elixir that does not require some runtime configuration / preparation... In the end I decided to stick with less magic and a bit cumbersome pattern of taking the module name from application env with default fallback... 

https://github.com/happycodrz/flexi/blob/eed717ceaa1396c4f60bea0514e2c65a3084b77f/lib/common.ex#L18

Test execution dropped from 3 seconds to 0.07 seconds... 

thanks and good luck!

**Comment by implicitly-awesome** [comment]
@mindreframer 
Yes, I'm familiar with this approach. Simply resolve modules in config and be happy if you have a few of such mocks. Once you get a system bigger and many modules to mock: you will end up with all modules redefined in the application configs. I've faced the issue when in an application the team follows service object (module) pattern and has a lot of such services modules.
It is just another approach with the most less overhead (in terms of performance), but sometimes it is hard to follow.

Swap's approach is mostly for testing purposes when you don't want to deal with configs, behaviours and all this stuff that (obviously) is faster. I tried to invent another alternative to https://github.com/jjh42/mock (which sometimes really hard to read), https://github.com/plataformatec/mox (which needs additional configuration and pre-defined behaviours) and "sandbox" (like that you've shared above).

Anyway, thank you for the feedback, I will think about the performance.
